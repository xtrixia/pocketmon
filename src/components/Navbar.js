import clsx from "clsx";
import { css } from "@emotion/css";
import { useHistory, useLocation, withRouter } from "react-router-dom";

import Button from "./Button";

import { COLORS } from "../root/colors";
import { SPACINGS } from "../root/spacings";
import { BREAKPOINTS } from "../root/breakpoints";

const topHeader = css`
  background: ${COLORS.terniary};
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
  display: flex;
  align-items: center;
  padding: ${SPACINGS.sm} ${SPACINGS.lg};
  @media (min-width: ${BREAKPOINTS.sm}) {
    justify-content: space-between;
  }
`;

const logo = css`
  flex-grow: 1;
  padding: 0 ${SPACINGS.xxl};
  cursor: pointer;
  @media (min-width: ${BREAKPOINTS.sm}) {
    flex-grow: 0;
    width: 350px;
  }
`;

const logoCenter = css`
  justify-content: center;
`;

const cta = css`
  width: 15%;
  @media (min-width: ${BREAKPOINTS.sm}) {
    cursor: pointer;
    width: 10%;
  }
  @media (min-width: ${BREAKPOINTS.lg}) {
    width: 5%;
  }
`;

function Navbar() {
  const history = useHistory();
  const location = useLocation();

  const isHomePage = location.pathname === "/";
  const isMyPokemonListPage = location.pathname === "/pocket";

  const handleRoute = () => {
    const currentPath = location.pathname;
    switch (currentPath) {
      case "/pocket":
        history.goBack();
        break;
      case "/list":
        history.push("/");
        break;
      default:
        history.push("/list");
        break;
    }
  };

  return (
    <header
      className={css`
        position: sticky;
        top: 0;
        z-index: 1;
      `}
    >
      <div className={clsx(topHeader, isHomePage && logoCenter)}>
        {!isHomePage && (
          <Button onClick={handleRoute}>
            <img
              src="https://ik.imagekit.io/xtrixia/Pocketmon/left-arrow_J_B2Zp65M.svg"
              alt="Left arrow"
              width="16px"
              height="16px"
            />
          </Button>
        )}

        <img
          loading="lazy"
          src="https://ik.imagekit.io/xtrixia/PoCKéTMoN_K8wj-F3Wm.png"
          srcSet="https://ik.imagekit.io/xtrixia/PoCKéTMoN_K8wj-F3Wm.png 200w,
          https://ik.imagekit.io/xtrixia/PoCKéTMoN_K8wj-F3Wm.png 400w"
          sizes="(max-width: 400px) 200px,
            400px"
          alt="Pockétmon logo"
          width="200px"
          height="80px"
          role="button"
          className={clsx(
            logo,
            css`
              margin: ${isMyPokemonListPage && "0 auto"};
            `
          )}
          onClick={() => history.push("/")}
          tabIndex={0}
        />

        {!isHomePage && !isMyPokemonListPage && (
          <img
            src="https://ik.imagekit.io/xtrixia/Pocketmon/pokeballs_zLTlXO-5_wx.svg"
            alt="My Pokémon list"
            width="45px"
            role="button"
            className={cta}
            onClick={() => history.push("/pocket")}
          />
        )}
      </div>
    </header>
  );
}
export default withRouter(Navbar);
