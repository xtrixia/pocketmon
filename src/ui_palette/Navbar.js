import clsx from "clsx";
import { css } from "@emotion/css";
import { useHistory, useLocation, withRouter } from "react-router-dom";

import LeftArrowSVG from "../assets/left-arrow.svg";
import LogoSVG from "../assets/logo.svg";
import PokeBallsSVG from "../assets/pokeballs.svg";

import { COLORS } from "../root/colors";
import { SPACINGS } from "../root/spacings";
import { BREAKPOINTS } from "../root/breakpoints";

const topHeader = css`
  background: ${COLORS.terniary};
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
  display: flex;
  align-items: center;
  padding: ${SPACINGS.sm};
  @media (min-width: ${BREAKPOINTS.sm}) {
    justify-content: space-between;
  }
`;

const logo = css`
  flex-grow: 1;
  padding: 0 ${SPACINGS.xxl};
  cursor: pointer;
  @media (min-width: ${BREAKPOINTS.sm}) {
    flex-grow: 0;
    width: 350px;
  }
`;

const logoCenter = css`
  justify-content: center;
`;

const cta = css`
  width: 15%;
  @media (min-width: ${BREAKPOINTS.sm}) {
    cursor: pointer;
    width: 10%;
  }
  @media (min-width: ${BREAKPOINTS.lg}) {
    width: 5%;
  }
`;

const backButton = css`
  padding: ${SPACINGS.xxs};
  background: ${COLORS.white};
  box-shadow: 1px 2px;
  border-radius: 5px;
  border: 1px solid ${COLORS.dark};
`;

function Navbar() {
  const history = useHistory();
  const location = useLocation();

  const isHomePage = location.pathname === "/";

  const handleRoute = () => {
    const currentPath = location.pathname;
    switch (currentPath) {
      case "/pocket":
        history.goBack();
        break;
      case "/list":
        history.push("/");
        break;
      default:
        history.push("/list");
        break;
    }
  };

  return (
    <header>
      <div className={clsx(topHeader, isHomePage && logoCenter)}>
        {!isHomePage && (
          <button className={backButton} onClick={handleRoute}>
            <img src={LeftArrowSVG} alt="Left arrow" width="16px" />
          </button>
        )}

        <img
          src={LogoSVG}
          alt="Pockétmon logo"
          width="200px"
          role="button"
          className={logo}
          onClick={() => history.push("/")}
          tabIndex={0}
        />

        {!isHomePage && (
          <img
            src={PokeBallsSVG}
            alt="My Pokémon list"
            width="45px"
            role="button"
            className={cta}
            onClick={() => history.push("/pocket")}
          />
        )}
      </div>
    </header>
  );
}
export default withRouter(Navbar);
