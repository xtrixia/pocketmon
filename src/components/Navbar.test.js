import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import Navbar from "./Navbar";

test("Navbar should render correctly", () => {
  const component = renderer.create(
    <Router>
      <Navbar />
    </Router>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.type).toEqual("header");
  // check if logo exist
  expect(tree.children[0].children[0].type).toEqual("img");
  expect(tree.children[0].children[0].props.alt).toEqual("Pock√©tmon logo");
});
