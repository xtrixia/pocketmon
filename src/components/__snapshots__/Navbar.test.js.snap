// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<header
  className="css-1dfxws0"
>
  <div
    className="css-1y2wuqc css-xf7v2y"
  >
    <img
      alt="Pockétmon logo"
      className="css-1uu0sh9 css-1dn73mw"
      height="80px"
      loading="lazy"
      onClick={[Function]}
      role="button"
      sizes="(max-width: 400px) 200px, 400px"
      src="https://ik.imagekit.io/xtrixia/PoCKéTMoN_K8wj-F3Wm.png"
      srcSet="https://ik.imagekit.io/xtrixia/PoCKéTMoN_K8wj-F3Wm.png 200w, https://ik.imagekit.io/xtrixia/PoCKéTMoN_K8wj-F3Wm.png 400w"
      tabIndex={0}
      width="200px"
    />
  </div>
</header>
`;
